File: seung365/programmers/10.py
def solution(s):
    answer = 0
    s=[i for i in s]
    
    for i in range(len(s)):
        
        stack = [] 
        
        check = True
        for char in s:
            if(char == '('):
                stack.append(char)
            elif(char == ')'):
                if stack and stack[-1] == '(':
                    stack.pop()
                else:
                    check = False
                    break
            elif(char == '['):
                stack.append(char)
            elif(char == ']'):
                if stack and stack[-1] == '[':
                    stack.pop()
                else:
                    check = False
                    break
            elif(char == '{'):
                stack.append(char)
            elif(char == '}'):
                if stack and stack[-1] == '{':
                    stack.pop()
                else:
                    check = False
                    break
        
        if check and not stack:
            answer += 1
        
        s.append(s.pop(0))
        
        
        
    
    
    return answer

File: jijun0129/Algorithm/10.js
function solution(s) {
  var str = [...s];
  var answer = 0;
  for (var i = 0; i < str.length; i++) {
    if (equalCheck(str)) {
      answer += 1;
    }
    var str1 = str.slice(1);
    var str2 = str.slice(0, 1);
    str1.push(str2[0]);
    str = [...str1];
  }
  return answer;
}

function equalCheck(str) {
  var stack = [];

  for (const ch of str) {
    if (ch === "[" || ch === "{" || ch === "(") {
      stack.push(ch);
    } else if (ch === "]" || ch === "}" || ch === ")") {
      if (stack.length === 0) {
        return false;
      } else {
        const check = stack.pop();
        if (ch === "]" && (check === "{" || check === "(")) {
          return false;
        } else if (ch === "}" && (check === "[" || check === "(")) {
          return false;
        } else if (ch === ")" && (check === "[" || check === "{")) {
          return false;
        }
      }
    }
  }
  if (stack.length === 0) {
    return true;
  }
}
